external-crates/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -209,13 +374,13 @@ impl Interpreter {
 
         let link_context = data_store.link_context();
         let mut current_frame = self
-            .make_new_frame(link_context, loader, function, ty_args, locals)
+            .make_new_frame(function, ty_args.clone(), locals)
             .map_err(|err| self.set_location(err))?;
         loop {
             let resolver = current_frame.resolver(link_context, loader);
             let exit_code =
                 current_frame //self
-                    .execute_code(&resolver, &mut self, data_store, gas_meter)
+                    .execute_code(&resolver, &mut self, plugins, data_store, gas_meter)
```

Comment:
should the `plugins` live in the `Interpreter` so we do not pass it around as much?

Reply:
It may be because I'm still a noob at Rust, but I can't seem to get this to work. If I nest plugins in Interpreter, then try to pass Interpreter as an immutable reference &dyn InterpreterInterface to plugins while mutating plugins, Rust complains about this simultaneous mutable and immutable borrow.

Reply:
I'll think about this more

Reply:
did you sort this out? I have hard time imaging what could go wrong if we did that

Reply:
There's nothing necessarily wrong with moving the plugins into the Interpreter struct.But we'll run into Rust borrow checker complaints for example in execute_code_impl. Today, we pass the following:
```
                Interpreter::pre_instr(
                    interpreter,
                    plugins,
                    gas_meter,
                    &self.function,
                    instruction,
                    &self.locals,
                    &self.ty_args,
                    resolver,
                )?;
```
And this is to circumvent Rust borrow checker complaining about 1 mut borrow + any other borrows. With plugins on Interpreter, I would be inclined to do something like:
```
    pub fn pre_instr_v2(
        &mut self,        
        gas_meter: &mut impl GasMeter,
        function: &Function,
        instruction: &Bytecode,
        locals: &Locals,
        ty_args: &[Type],
        resolver: &Resolver,
    ) -> PartialVMResult<()> {
        for plugin in self.plugins.iter_mut() {
            let result = plugin.pre_instr(
                self,
                function,
                instruction,
                locals,
                ty_args,
                resolver,
            );
            if plugin.is_critical() {
                result?;
            }
        }

        profile_open_instr!(gas_meter, format!("{:?}", instruction));
        Ok(())
    }
```
But then because we mutably borrow plugins, then self (interpreter) to use some read-only methods, Rust complains. I suppose I could continue to pass interpreter and plugins in separately, and doing this would still reduce the number of times we need to pass plugin around? Might just be confusing since we don't use self.plugins